cockatrice/src/chatview.cpp

```diff
@@ -73,6 +74,14 @@ void ChatView::appendHtml(const QString &html)
         verticalScrollBar()->setValue(verticalScrollBar()->maximum());
 }
 
+void ChatView::appendHtmlServerMessage(const QString &html)
+{
+    bool atBottom = verticalScrollBar()->value() >= verticalScrollBar()->maximum();
+    prepareBlock().insertHtml("<font color=" + SERVER_MESSAGE_COLOR + ">" + html + "</font>");
+    if (atBottom)
+        verticalScrollBar()->setValue(verticalScrollBar()->maximum());
+}
```

Comment:
This function can be replace with `appendHtml("<font ..." + html + "</font>")` instead of re implementing the logic